x/ccv/provider/keeper/msg_server.go

```diff
@@ -309,16 +293,55 @@ func (k msgServer) SetConsumerCommissionRate(goCtx context.Context, msg *types.M
 	return &types.MsgSetConsumerCommissionRateResponse{}, nil
 }
 
-func (k msgServer) ConsumerModification(goCtx context.Context, msg *types.MsgConsumerModification) (*types.MsgConsumerModificationResponse, error) {
-	if k.GetAuthority() != msg.Authority {
-		return nil, errorsmod.Wrapf(types.ErrUnauthorized, "expected %s, got %s", k.GetAuthority(), msg.Authority)
+// RegisterConsumer registers a consumer chain
+func (k msgServer) RegisterConsumer(goCtx context.Context, msg *types.MsgRegisterConsumer) (*types.MsgRegisterConsumerResponse, error) {
```

Comment:
Do we need to set defaults for the rest of the consumer state? Or are the getter good enough to return defaults when no state was set?

Reply:
Not sure I follow. What is the rest of the consumer state? You refer to the initialization and update records?
If yes, I do not see why we set them at this stage. They would be set in due time during the `MsgInitializeConsumer` and `MsgUpdateConsumer` messages.

Reply:
What happens if a user is sending a MsgInitializeConsumer without sending a MsgUpdateConsumer? Would the consumer start without power shaping configs? What would that even mean?